agent/engine/serviceconnect/version_linux.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package serviceconnect
+
+import (
+	"fmt"
+	"os"
+	"sort"
+)
+
+const (
+	appnetInterfaceV1          = "v1"
+	serviceConnectCapabilityV1 = "ecs.service-connect.v1"
+)
+
+var (
+	supportedAppnetInterfaceVerToCapability = map[string][]string{
+		appnetInterfaceV1: {
+			serviceConnectCapabilityV1,
+		},
+	}
+)
+
+// getSupportedAppnetInterfaceVersion returns the latest supported AppNet interface version
+// by looking for the AppNet agent tar name available on the instance
+// from supportedAppnetInterfaceVerToCapability map from highest to lowest version
+func getSupportedAppnetInterfaceVersion() string {
+	supportedAppnetInterfaceVersions := make([]string, 0, len(supportedAppnetInterfaceVerToCapability))
+	for supportedAppnetInterfaceVersion := range supportedAppnetInterfaceVerToCapability {
+		supportedAppnetInterfaceVersions = append(supportedAppnetInterfaceVersions, supportedAppnetInterfaceVersion)
+	}
+	sort.Sort(sort.Reverse(sort.StringSlice(supportedAppnetInterfaceVersions)))
+
```

Comment:
I think it would be better if we return the list right here, and let the loader go through the list and validate the existence. So something like
```
for each appnetInterfaceVersion in getSupportedAppnetInterfaceVersions():  // from highest to lowest
  if tar file does not exist, continue
  if loading tar file failed, continue
  # if we reach here we've successfully loaded a tar file
  break
```
That's because
- it will be a cleaner separation between static capability and runtime validation (which needs to be performed by loader anyway)
- there might be cases where the tar file exists but is corrupted, and in such case we should let loader keeping trying older version of AppNet